src/VisualStudio/Core/Def/Implementation/SymbolSearch/SymbolSearchResult.cs

```diff
@@ -0,0 +1,232 @@
+        {
+            return await MakeResultAsync(context, documentSpan, referenceItem: null, definitionItem: definition, cancellationToken: token).ConfigureAwait(false);
+        }
+
+        internal static async Task<RoslynSymbolSearchResult> MakeAsync(SymbolSearchContext context, SourceReferenceItem reference, DocumentSpan documentSpan, CancellationToken token)
+        {
+            return await MakeResultAsync(context, documentSpan, referenceItem: reference, definitionItem: reference.Definition, cancellationToken: token).ConfigureAwait(false);
+        }
+
+        private static async Task<RoslynSymbolSearchResult> MakeResultAsync(SymbolSearchContext context, DocumentSpan documentSpan, SourceReferenceItem referenceItem, DefinitionItem definitionItem, CancellationToken cancellationToken)
+        {
+            var (projectGuid, documentGuid, projectName, sourceText) = await FindUsagesUtilities.GetGuidAndProjectNameAndSourceTextAsync(documentSpan.Document, cancellationToken)
+                .ConfigureAwait(false);
+
+            var (excerptResult, lineText) = await FindUsagesUtilities.ExcerptAsync(sourceText, documentSpan, cancellationToken).ConfigureAwait(false);
+            var plainText = excerptResult.Content.ToString();
+            var result = new RoslynSymbolSearchResult(context, plainText);
+
+            result.ProjectGuid = projectGuid.ToString();
+            result.DocumentGuid = documentGuid.ToString();
+            result.ProjectName = projectName;
+
+            if (referenceItem != null)
+            {
+                var classifiedSpansAndHighlightSpan = await ClassifiedSpansAndHighlightSpanFactory.ClassifyAsync(referenceItem.SourceSpan, cancellationToken).ConfigureAwait(false);
+                var classifiedSpans = classifiedSpansAndHighlightSpan.ClassifiedSpans;
+                var docText = await referenceItem.SourceSpan.Document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                result.ContextText = new ClassifiedTextElement(classifiedSpans.Select(cspan => new ClassifiedTextRun(cspan.ClassificationType, docText.ToString(cspan.TextSpan))));
+                result.ContextHighlightSpan = AsSpan(classifiedSpansAndHighlightSpan.HighlightSpan);
+
+                if (referenceItem.AdditionalProperties.TryGetValue(AbstractReferenceFinder.ContainingTypeInfoPropertyName, out var containingTypeInfo))
+                {
+                    result.ContainingTypeName = containingTypeInfo;
+                }
+
+                if (referenceItem.AdditionalProperties.TryGetValue(AbstractReferenceFinder.ContainingMemberInfoPropertyName, out var containingMemberInfo))
+                {
+                    result.ContainingMemberName = containingMemberInfo;
+                }
+
+                // Get read\written information
+                if (referenceItem.IsWrittenTo)
```

Comment:
@mavasani Why do we still have both SourceReferenceItem.IsWrittenTo/IsReadFrom and also SourceReferenceItem.SymbolUsageInfo.IsWrittenTo()/IsReadFrom()? This looks like we can have the former but not the latter? If so, what's the difference?

Reply:
And if it's for some back compat for IVT reasons, can we just obsolete [IsWrittenTo] and have it just internally forward to SymbolUsageInfo.IsWrittenTo?